server.js

```diff
@@ -16,8 +24,97 @@ const welcomeMessage = {
 //Note: messages will be lost when Glitch restarts our server.
 const messages = [welcomeMessage];
 
-app.get("/", function (request, response) {
+
+let maxID = Math.max(...messages.map(c => c.id))
+
+app.get("/", function (request, response, next) {
   response.sendFile(__dirname + "/index.html");
 });
 
-app.listen(process.env.PORT);
+// app.get("/messages", function (request, response) {
+//   response.send(welcomeMessage);
+// });
+
+
+app.post('/messages', (req, res,next) => {
+  if (!req.body.from || !req.body.text){   
+    res.status(400).send(`You must include  name and message in your request`)
+    return
```

Comment:
Good use of return after sending a response to make sure you don't send another one